airsonic-main/src/main/java/org/airsonic/player/service/SecurityService.java

```diff
@@ -347,15 +395,111 @@ protected boolean isFileInFolder(String file, String folder) {
         return file.toUpperCase().startsWith(folder.toUpperCase());
     }
 
-    public void setSettingsService(SettingsService settingsService) {
-        this.settingsService = settingsService;
+
+    // =======================================================================================================
+    // Utilities for Sonos link.
+
+    public Authentication authenticate(String username, String password) {
+        UsernamePasswordAuthenticationToken authReq = new UsernamePasswordAuthenticationToken(username, password);
+        return authenticationManager.authenticate(authReq);
     }
 
-    public void setUserDao(UserDao userDao) {
-        this.userDao = userDao;
+    public void authenticate(String sonosLinkToken) throws SonosSoapFault.LoginUnauthorized {
+        SonosLink sonosLink = jwtSecurityService.verifySonosLink(sonosLinkToken);
```

Comment:
There is an authenticator for jwts, jwtauthenticationprovider, why not use that?

Or inject an authmanager.

Should we be using the jwt service directly when it has no concept of auth by itself? (it verifies and decodes jwt, without any notion of auth)